src/similarity.spec.ts

```diff
@@ -278,33 +340,82 @@ describe("similarity", () => {
             }),
           },
           response: DUMMY_RESPONSE,
-        }
+        },
+        new RewriteRules()
       )
     ).toBe(6);
+
+    // The following payloads are identical after rewrite rules have been applied.
+    expect(
+      computeSimilarity(
+        "POST",
+        "/test",
+        {},
+        compactJsonBuffer({
+          name: "Jane Doe",
+          email:
+            "jane.doe-some-test-6f82fbbe-d36a-4c5c-b47b-84100122fbbc@example.com",
+          age: 42,
+        }),
+        {
+          request: {
+            method: "POST",
+            path: "/test",
+            headers: {},
+            body: wellFormattedJsonBuffer({
+              name: "Jane Doe",
+              email: "jane.doe-some-test@example.com",
+              age: 42,
+            }),
+          },
+          response: DUMMY_RESPONSE,
+        },
+        new RewriteRules().appendRule(
+          new RewriteRule(
+            /-[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}(@example.com)$/gi,
+            "$1"
+          )
+        )
+      )
+    ).toBe(0);
```

Comment:
This is a new test that tests similarity with rewrite rules. The rest of the changes in this file are just formatting changes applied by the linter.